cloud-consumption/src/main/java/com/sequenceiq/consumption/events/ConsumptionStructuredFlowEventFactory.java

```diff
@@ -0,0 +1,58 @@
+package com.sequenceiq.consumption.events;
+
+import static com.sequenceiq.cloudbreak.structuredevent.event.StructuredEventType.FLOW;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.common.service.Clock;
+import com.sequenceiq.cloudbreak.structuredevent.event.CloudbreakEventService;
+import com.sequenceiq.cloudbreak.structuredevent.event.FlowDetails;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPOperationDetails;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPStructuredFlowEvent;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.freeipa.CDPFreeIpaStructuredFlowEvent;
+import com.sequenceiq.cloudbreak.structuredevent.service.CDPStructuredFlowEventFactory;
+import com.sequenceiq.consumption.domain.Consumption;
+import com.sequenceiq.consumption.service.ConsumptionService;
+import com.sequenceiq.flow.ha.NodeConfig;
+
+@Component
+public class ConsumptionStructuredFlowEventFactory implements CDPStructuredFlowEventFactory {
+
+    @Inject
+    private Clock clock;
+
+    @Inject
+    private ConsumptionService consumptionService;
+
+    @Inject
+    private NodeConfig nodeConfig;
+
+    @Value("${info.app.version:}")
+    private String serviceVersion;
+
+    @Override
+    public CDPStructuredFlowEvent createStructuredFlowEvent(Long resourceId, FlowDetails flowDetails) {
+        return createStructuredFlowEvent(resourceId, flowDetails, null);
+    }
+
+    @Override
+    public CDPStructuredFlowEvent createStructuredFlowEvent(Long resourceId, FlowDetails flowDetails, Exception exception) {
+        Consumption consumption = consumptionService.findConsumptionById(resourceId);
+        String resourceType = CloudbreakEventService.CONSUMPTION_RESOURCE_TYPE;
+        CDPOperationDetails operationDetails = new CDPOperationDetails(clock.getCurrentTimeMillis(), FLOW, resourceType, consumption.getId(),
+                consumption.getName(), nodeConfig.getId(), serviceVersion, consumption.getAccountId(), consumption.getResourceCrn(),
+                ThreadBasedUserCrnProvider.getUserCrn(), null, null);
+
+        CDPFreeIpaStructuredFlowEvent event = new CDPFreeIpaStructuredFlowEvent(operationDetails, flowDetails, null,
```

Comment:
You are returning a `CDPFreeIpaStructuredFlowEvent` tyoe

Reply:
Thanks, created CDPConsumptionStructuredFlowEvent